clickhouse-http-client/src/main/java/com/clickhouse/client/http/ClickHouseHttpConnection.java

```diff
@@ -206,6 +223,11 @@ protected static Map<String, String> createDefaultHeaders(ClickHouseConfig confi
             map.put(name, value);
         }
 
+        if (!LOCAL_ADDRESS.isEmpty() && !map.containsKey("referer"))
+        {
+            map.put("referer", LOCAL_ADDRESS);
```

Comment:
In some cases `referer` would be an application name. For example, in the cloud with dynamic IP addresses.

Reply:
I mean that some clients want to use a site name or application. As an example, when application run in k8s IP address would be hard to track back - so pod ID would be more handy.